### ISSUE TITLE
Checkout query parameters should be configurable

### ISSUE THREAD

**Issue by ezekg** [keygen-sh/keygen-go#11]
E.g. `ttl` and `include`. Currently these are hard-coded and some users may want to configure them differently.

**Comment by ezekg** [comment]
Design pattern for options: https://commandcenter.blogspot.com/2014/01/self-referential-functions-and-design.html

```go
func (m *Machine) Checkout(options ...func(CheckoutOptions) error) (*MachineFile, error) {
  opts := CheckoutOptions{Encrypt: true, Include: "license,license.entitlements"}
  for _, opt := range options {
    err := opt(opts)
    if err != nil {
      return nil, err
    }
  }

  // ...
}
```

**Comment by ezekg** [comment]
This would require that we define other include relationship types, e.g. `product`, `policy`, etc.:

https://github.com/keygen-sh/keygen-go/blob/d6911045cd386479c3aefaa7893404f73542ce91/machine_file.go#L149-L167

**Comment by ezekg** [comment]
Closed in 261833ff47104c9055299bac5b48dcdabe3c8942.
